src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -247,6 +250,31 @@ private boolean isTrackingOngoing() {
         return trip.journeyState.tripStatus == TripStatus.TRIP_ACTIVE && TripTrackingData.getOngoingTrackedJourney(trip.id) != null;
     }
 
+    /**
+     * Process leg transition notifications.
+     */
+    public void processLegTransition(List<NotificationType> legTransitionTypes, TravelerPosition travelerPosition) {
+        OtpUser otpUser = getOtpUser();
+        if (otpUser != null) {
+            otpUser.relatedUsers.forEach(relatedUser -> {
+                if (relatedUser.isConfirmed()) {
+                    OtpUser observer = Persistence.otpUsers.getOneFiltered(eq("email", relatedUser.email));
+                    if (observer != null) {
+                        enqueueNotification(
```

Comment:
I think we need to send the notification once, similar to the bus driver notifications. Otherwise, if the traveler stays at an origin or a transition spot for a while, we will be sending the same notification email every few seconds...

Reply:
The reason for using `CheckMonitoredTrip` is bc in `sendNotifications(OtpUser otpUser)` there is a check to not send notifications if already sent `if (previousJourneyState.lastNotifications.containsAll(notifications) && !hasInitialReminder)`. Then I noticed that `lastNotifications` is never set! So, I have made an update in this area. The advantage of this approach is that the notification will be sent again if not successful the first time.

Reply:
Updated in: https://github.com/ibi-group/otp-middleware/pull/275/commits/91d66190cb611283993b8e745148641efd695c5e